server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -1095,9 +754,12 @@ public static int failedShards(ImmutableOpenMap<ShardId, RestoreInProgress.Shard
         return failedShards;
     }
 
-    private static Map<String, String> renamedIndices(RestoreSnapshotRequest request, List<String> filteredIndices,
```

Comment:
Made this return `Map<String, IndexId>` instead right away so we don't have to keep a hold of the `RepositoryData` for the duration of the CS update request only to look up the `IndexId` later.